src/components/Search/SearchHeader.tsx

```diff
@@ -0,0 +1,90 @@
+import { useSearchStore } from '~/components/Search/useSearchState';
+import { useInstantSearch, useSearchBox } from 'react-instantsearch-hooks-web';
+import {
+  createStyles,
+  Group,
+  SegmentedControl,
+  SegmentedControlItem,
+  Text,
+  ThemeIcon,
+  Title,
+} from '@mantine/core';
+import { IconCategory, IconFileText } from '@tabler/icons-react';
+import { useRouter } from 'next/router';
+import { removeEmpty } from '~/utils/object-helpers';
+
+const useStyles = createStyles((theme) => ({}));
```

Comment:
Seems like we are only using this to get the theme, we can replace it with `useMantineTheme`
```suggestion
```